pynamodb/attributes.py

```diff
@@ -40,11 +40,13 @@ def __init__(self,
 
     def __set__(self, instance, value):
         if instance:
-            instance.attribute_values[self.attr_name] = value
+            attr_name = instance._dynamo_to_python_attrs.get(self.attr_name, self.attr_name)
+            instance.attribute_values[attr_name] = value
 
     def __get__(self, instance, owner):
         if instance:
-            return instance.attribute_values.get(self.attr_name, None)
+            attr_name = instance._dynamo_to_python_attrs.get(self.attr_name, self.attr_name)
+            return instance.attribute_values.get(attr_name, None)
```

Comment:
I agree and I'm not a huge fan of all of the conversions we have to do between these. This is mostly a function of earlier design decisions made.

That said, I'm OK with this for now because it's still easy to identify all of the places that these things happen if we want to do a larger refactor later